### PR TITLE
Cumulative changes from gwpy-plot testing

### PR DESCRIPTION
**Pull request by areeda**
Sorry for the "merge" commits rebase failed before each one.
Annotation for changed files:

**bin/gwpy-plot**
* move test for no DISPLAY to top for matplotlib.use()

**gwpy/cli/cliproduct.py**
* :rtype annotation added to log() docs

**gwpy/cli/qtransform.py**
* `def fformat(x):  # float format` failed if passed a str arg

**gwpy/cli/spectrogram.py**
* Deal with zeroes in resulting spectrogram and log color scale

### CODE REVIEW

File: gwpy/cli/qtransform.py [hunk 1]
```diff
@@ -148,6 +149,9 @@ def fformat(x):  # float format
                 return '[{0}]'.format(', '.join(map(fformat, x)))
             if isinstance(x, Quantity):
                 x = x.value
+            elif isinstance(x, str):
```
Comment (duncanmmacleod) [inline gwpy/cli/qtransform.py]: how does `fformat` ever get called with a `str`? if its via `qrange` I think a better switch would be to set `type=float` in the `--qrange` option definition in line 101.

Comment (areeda) [reply gwpy/cli/qtransform.py]: This is one of your changes. I needed this to go further.
I'll look into it.

Comment (duncanmmacleod) [reply gwpy/cli/qtransform.py]: thanks @areeda, I should have been a bit clearer, reading back my message it sounds rude, sorry. I'm just surprised that this hasn't been caught before, can you post a command-line that reproduces it?

**Comment by duncanmmacleod** [comment]
@areeda, is there anything else to go into this PR?

**Pull request merged**